src/tools/pub/cmd/pub-add.md

```diff
@@ -6,7 +6,7 @@ description: Use dart pub add to add a dependency.
 _Add_ is one of the commands of the [pub tool](/tools/pub/cmd).
 
 ```nocode
-$ dart pub add [dev:]<package>[:<constraint>] [[dev:]<package>[:<constraint>]... ] [options]
+$ dart pub add [{dev|override}:]<package>[:descriptor] [[{dev|override}:]<package>[:descriptor] ...] [options]
```

Comment:
My understanding is just one type of descriptor. It just often looks simpler since there is no nested information. You can also include `version` as part of the JSON descriptor:

`'test:{"hosted":"https://pub.dev","version":"1.24.1"}'`

Reply:
Ok, so descriptor includes constraint and other things. Makes sense then, good change